### PR TITLE
refactor: replace axios with fetch

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
The motivation is primarily performance: fetch already exists in the browser, but axios does not.  So, this is one less package in the bundle.

```js
credentials: environment === 'development' ? 'include' : 'same-site'
```

This probably just needs to be `include`, but I'd like to try `same-site` on staging.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
FYI: https://caniuse.com/?search=fetch 96%, so that's probably fine.  Includes recent versions of Safari, too.

**Comment by ojeytonwilliams** [comment]
Re: `same-site` vs `include` it's probably just that the client/server ports are different locally and so they count as different sites for fetch.

**Comment by ahmadabdolsaheb** [comment]
I think we are increasing the security with this pr as well. So as you suggested we can space out deployments to isolate any issues that might come up.

**Review by ahmaxed — DISMISSED** [review]

File: client/src/utils/get-words.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
-const config = require('../../../config/env.json');
+import env from '../../../config/env.json';
 
-const words = require(`../../i18n/locales/${config.clientLocale}/motivation.json`);
+import english from '../../i18n/locales/english/motivation.json';
+import espanol from '../../i18n/locales/espanol/motivation.json';
+import chinese from '../../i18n/locales/chinese/motivation.json';
+
+// The idea here is to use static imports to allow for tree-shaking and dead
```
Comment (ahmadabdolsaheb) [inline client/src/utils/get-words.js:7]: Is this part of this pr? it seems like i18n related.

**Review by ahmaxed — DISMISSED** [review]

File: client/src/utils/get-words.js (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/utils/get-words.js:7]: Ahem.  I'll remove it.

File: client/src/utils/ajax.js [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
+  const options = {
+    ...defaultOptions,
+    method,
+    headers: {
+      'CSRF-Token': getCSRFToken(),
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  };
+  return fetch(`${base}${path}`, options).then(res => res.json());
```
Comment (nhcarrigan) [inline client/src/utils/ajax.js:44]: Okay this took a bit to debug, but...

Certain endpoints we use don't actually send back JSON, such as the `/account/reset-progress` endpoint. When attempting to perform this action, the `res.json()` call throws an error and we get the "Something went wrong" flash message.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/src/utils/ajax.js (hunk 2 above)
Comment (nhcarrigan) [reply client/src/utils/ajax.js:44]: For clarification, this was the response and result of res.json():

image

For testing, I changed it to `res.text()`:

image

**Review by ShaunSHamilton — COMMENTED** [review]
Well, I am unsure what might be the issue on my side, but I was unable to:
- Reset account progress
- Change my email
- Reporting a user (i suspect)

Most everything else worked as expected.

File: client/src/utils/ajax.js (hunk 2 above)
Comment (nhcarrigan) [reply client/src/utils/ajax.js:44]: @ojeytonwilliams what's the performance increase here? Is it worth the overhead of potentially refactoring our API to always send JSON?

**Comment by ojeytonwilliams** [comment]
> Well, I am unsure what might be the issue on my side, but I was unable to:
> 
> * Reset account progress
> * Change my email
> * Reporting a user (i suspect)

For anything email related, you'll need something to intercept the emails (like [Mailhog](https://contribute.freecodecamp.org/#/how-to-catch-outgoing-emails-locally?id=installing-mailhog)), otherwise the api will (correctly) throw an error.

I'm looking into how to properly test these things in CI.

File: client/src/utils/ajax.js (hunk 2 above)
Comment (ojeytonwilliams) [reply client/src/utils/ajax.js:44]: Nice job catching and debugging this.  I need to be more careful in future.

The decrease in the bundle is fairly modest.  However I think unifying how the API responds is worthwhile in and of itself, so I'll take a look.  Before I do any of that, though, I'm going to add some more tests!

**Comment by raisedadead** [comment]
> All being well this will fail with flying colours.

THAT!

**Comment by nhcarrigan** [comment]
image

A beautiful sight!

**Comment by ojeytonwilliams** [comment]
To give a little more context for the last commit https://github.com/freeCodeCamp/freeCodeCamp/pull/41944/commits/ef3e7c676f947efcae33ba994a7fb899c9020fcc the rationale was that I don't think we will ever want to do anything other than redirect and/or return some json.

**Comment by ahmadabdolsaheb** [comment]
For reviewing this, I think hitting all of the api points from the UI should suffice. Any thing else comes to mind?

**Comment by ojeytonwilliams** [comment]
@raisedadead pointed out in chat that this can't be deployed as is because the updated client does not support all the responses from the existing api.  We cannot guarantee that the updated client talks to the updated api during deployment.

A tentative plan is to deploy in stages.

1. If necessary, update the client to handle both the existing responses and the new, json-only responses
2. Update the api to only respond with json
3. Update the client to only accept json.

**Comment by ojeytonwilliams** [comment]
https://github.com/freeCodeCamp/freeCodeCamp/pull/42320 is live in production, so I think this is ready for another round of reviews.

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍

**Review by raisedadead — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
This looks good to me. Couldn't replicate the same errors as before.

**Pull request merged**